src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -164,25 +164,25 @@ public sealed class LegacyPackageReferenceProject : BuildIntegratedNuGetProject
 
         #endregion
 
-        private string GetBaseIntermediatePath(bool shouldThrow = true)
+        private string GetRestoreOutputPath(bool shouldThrow = true)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            var baseIntermediatePath = _vsProjectAdapter.BaseIntermediateOutputPath;
+            var restoreOutputPath = _vsProjectAdapter.RestoreOutputPath;
```

Comment:
I think this should be MSBuildProjectExtensionsPath, or did we misunderstand each other?

RestoreOutputPath is only meant to be used in non-VS/non-build scenarios. Namely the tool restore. 

The property here should be the same as the one we read in MsBuild.